pycbc/inference/models/hierarchical.py

```diff
@@ -599,3 +602,273 @@ def _loglikelihood(self):
             for stat in self.extra_stats_map[lbl]:
                 setattr(self._current_stats, stat, mstats[stat.subname])
         return logl
+
+
+class MultibandRelativeTimeDom(HierarchicalModel):
```

Comment:
It's not really time to the RelativeTimeDom, right? It's just tied to a primary model / secondary model scenario, no?

Reply:
I named it this way because the primary model uses `RelativeTimeDom`, and also emphasizes the multiband purpose. I know you want a more general likelihood model though.

Reply:
@WuShichao I don't think this is tied to the relative model, so I think the name still needs to be fixed. (it is tied to models that do marginalization).

Reply:
@ahnitz Can you suggest a name? This multiband likelihood relies on the primary model's `RelativeTimeDom`, and has specific code to do the marginalization for LISA. So it's not a likelihood model for general usage, right?

Reply:
It doesn't is just relies on it doing vector maginalization. Nothing explicitly requires that either be a LISA model or even be specific models, just that the primary use vector marginalization at some point. 

It could be used for general useage if for example you want to do joint EM + GW analysis. In that case, sky marginalization would still be useful and one could just use your code with modified configuration files.

Reply:
@ahnitz OK, then let's name it `JointPrimaryModelMarginalized`, do you think it's better?